Improve memory manager docs (#49754)
kubernetes/website: content/en/docs/tasks/administer-cluster/memory-manager.md

for Pods in Guaranteed QoS class.
To immediately put the Memory Manager into operation follow the guidelines in the section
[Memory Manager configuration](#memory-manager-configuration), and subsequently,
prepare and deploy a `Guaranteed` Pod as illustrated in the section
[Placing a Pod in the Guaranteed QoS class](#placing-a-pod-in-the-guaranteed-qos-class).

The Memory Manager is a hint provider, and it provides topology hints for
the Topology Manager which then aligns the requested resources according to these topology hints.
On Linux, it also enforces `cgroups` (specifically, `cpuset.mems`) for Pods.
The complete flow diagram concerning pod admission and deployment process is illustrated
below:

![Memory Manager in the pod admission and deployment process](/images/docs/memory-manager-diagram.svg)

During this process, the Memory Manager updates its internal counters stored in
[Node Map and Memory Maps][2] to manage guaranteed memory allocation.

The memory manager activates during kubelet startup if a node administrator configures
`reservedMemory` for the kubelet (section [Reserved memory configuration](#reserved-memory-flag)).

> In this case, the kubelet updates its node map  to reflect this reservation.

Reviewer: ```suggestion
In this case, the kubelet updates its node map to reflect this reservation.
```

When the `Static` policy is configured, you **must** configure reserved memory for the node
(for example, with the `reservedMemory` configuration field in the kubelet configuration).

An important topic in the context of Memory Manager operation is the management of NUMA groups.
Each time pod's memory request is in excess of single NUMA node capacity, the Memory Manager
attempts to create a group that comprises several NUMA nodes and that features extended memory
capacity.

## Memory Manager configuration

Other Managers should already be configured (see [resource alignment prerequisites](#resource-alignment-prerequisites).
Set the `memoryManagerPolicy` configuration field within the [kubelet configuration]({{< relref "/docs/reference/config-api/kubelet-config.v1beta1" >}}), to the name of your chosen [policy](#policies).

Optionally, some amount of memory can be reserved for system or kubelet processes to increase
node stability (section [Reserved memory configuration](#reserved-memory-flag)).

### Policies

> Kubernetes' memory manager provides three policies. You can select a policy via the `memoryManagerPolicy` configuration field

Reviewer: ```suggestion
Kubernetes memory manager provides three policies. You can select a policy via the `memoryManagerPolicy` configuration field
```

Author: This would change the meaning; the new text isn't idiomatic.

Reviewer: Oh my bad! it was a mistake.

{{< feature-state feature_gate_name="WindowsCPUAndMemoryAffinity" >}}

**This policy is only supported on Windows.**

On Windows, NUMA node assignment works differently than Linux.
There is no mechanism to ensure that Memory access only comes from a specific NUMA node.
Instead the Windows operating system scheduler selects the most optimal NUMA node based on the CPU(s) assignments.

> It is possible that Windows might use other NUMA nodes if deemed optimal by the Windows scheduler.

Reviewer: ```suggestion
It is possible that Windows might use other NUMA nodes if the Windows scheduler deems them optimal.
```

The policy does track the amount of memory available and requested through the internal _node map_.
The memory manager makes a best effort at ensuring that enough memory is available on a NUMA node before making
a resource assignment.  
This means that in most cases memory assignment should function as specified.

## Reserved memory configuration {#reserved-memory-flag}

As an administrator, you can configure the total amount of reserved memory
for a node. This pre-configured value is subsequently utilized to calculate
the real amount of [node allocatable](/docs/tasks/administer-cluster/reserve-compute-resources/#node-allocatable) memory available to pods.

The Kubernetes scheduler incorporates allocatable memory information to optimise pod
[scheduling](/docs/concepts/scheduling-eviction/).
. The _node allocatable_ mechanism is commonly used by node administrators to reserve K8s node

> system resources for the kubelet or operating system processes to help assure node stability.

Reviewer: ```suggestion
system resources for the kubelet or operating system processes to help ensure node stability.
```

Author: Assure seems fine here; why change it?
